File: app/src/main/java/com/samyak/cowin_tracker/MyFirebaseMessagingService.java
package com.samyak.cowin_tracker;

import android.os.Build;
import android.util.Log;

import androidx.annotation.NonNull;
import androidx.annotation.RequiresApi;

import com.google.firebase.messaging.FirebaseMessagingService;
import com.google.firebase.messaging.RemoteMessage;


public class MyFirebaseMessagingService extends FirebaseMessagingService {

    private static final String TAG = "MyFirebaseMsgService";

    @RequiresApi(api = Build.VERSION_CODES.O)
    @Override
    public void onMessageReceived(@NonNull RemoteMessage remoteMessage) {
        super.onMessageReceived(remoteMessage);

        if (remoteMessage.getData().size() > 0) {
            Log.d(TAG, "Message data payload: " + remoteMessage.getData());
            String date = remoteMessage.getData().get("date");
            String pincode = remoteMessage.getData().get("pincode");
            String slottype = remoteMessage.getData().get("slot_type");


            if (remoteMessage.getNotification() != null) {
                Log.d(TAG, "Message Notification Title: " + remoteMessage.getNotification().getTitle());
                Log.d(TAG, "Message Notification Body: " + remoteMessage.getNotification().getBody());

//                Intent intent = new Intent(this, MainActivity.class);
//                intent.putExtra("date", date);
//                intent.putExtra("pincode", pincode);
//                intent.putExtra("slot_type", slottype);

            }


        }

    }

}

File: app/src/main/java/com/samyak/cowin_tracker/cloudmessaging/MyFirebaseMessagingService.kt
package com.samyak.cowin_tracker.cloudmessaging

import android.util.Log
import com.google.firebase.messaging.FirebaseMessagingService
import com.google.firebase.messaging.RemoteMessage
import com.samyak.cowin_tracker.TAG

class MyFirebaseMessagingService : FirebaseMessagingService() {

    override fun onMessageReceived(message: RemoteMessage) {
        super.onMessageReceived(message)

        Log.d(TAG, "onMessageReceived: $message")

    }

    override fun onNewToken(token: String) {
        super.onNewToken(token)
        Log.d(TAG, "onNewToken: $token")
    }

} 